smithy-swift-codegen/src/main/kotlin/software/amazon/smithy/swift/codegen/lang/Function.kt

```diff
@@ -20,17 +20,27 @@ data class Function(
     val accessModifier: AccessModifier = AccessModifier.Public,
     val isAsync: Boolean = false,
     val isThrowing: Boolean = false,
+    val isMutating: Boolean = false,
 ) {
 
     /**
      * Render this function using the given writer.
      */
     fun render(writer: SwiftWriter) {
+        val renderedMutating = "mutating ".takeIf { isMutating } ?: ""
         val renderedParameters = parameters.joinToString(", ") { it.rendered(writer) }
         val renderedAsync = if (isAsync) "async " else ""
         val renderedThrows = if (isThrowing) "throws " else ""
         val renderedReturnType = returnType?.let { writer.format("-> \$N ", it) } ?: ""
-        writer.openBlock("${accessModifier.renderedRightPad()}func $name($renderedParameters) $renderedAsync$renderedThrows$renderedReturnType{", "}") {
+        writer.openBlock(
+            "\$L\$Lfunc \$L(\$L) \$L$renderedThrows$renderedReturnType{",
+            "}",
+            accessModifier.renderedRightPad(),
+            renderedMutating,
+            name,
+            renderedParameters,
+            renderedAsync,
+        ) {
```

Comment:
Added the optional `mutating` modifier here & did a lot of code cleanup.

(`renderedThrows ` and `renderedReturnType` remain interpolated because `openBlock` takes a max number of arguments.)